doc/manual/src/language/values.md

```diff
@@ -92,39 +92,48 @@
 
 - <a id="type-path" href="#type-path">Path</a>
 
-  *Paths*, e.g., `/bin/sh` or `./builder.sh`. A path must contain at
-  least one slash to be recognised as such. For instance, `builder.sh`
-  is not a path: it's parsed as an expression that selects the
-  attribute `sh` from the variable `builder`. If the file name is
-  relative, i.e., if it does not begin with a slash, it is made
-  absolute at parse time relative to the directory of the Nix
-  expression that contained it. For instance, if a Nix expression in
-  `/foo/bar/bla.nix` refers to `../xyzzy/fnord.nix`, the absolute path
-  is `/foo/xyzzy/fnord.nix`.
-
-  If the first component of a path is a `~`, it is interpreted as if
-  the rest of the path were relative to the user's home directory.
-  e.g. `~/foo` would be equivalent to `/home/edolstra/foo` for a user
-  whose home directory is `/home/edolstra`.
-
-  For instance, evaluating `"${./foo.txt}"` will cause `foo.txt` in the current directory to be copied into the Nix store and result in the string `"/nix/store/<hash>-foo.txt"`.
-
-  Note that the Nix language assumes that all input files will remain _unchanged_ while  evaluating a Nix expression.
+  *Paths* are distinct from strings and can be expressed by path literals such as `./builder.sh`.
+
+  Paths are the preferred type for referring to local files.
+  This is thanks to the following properties:
+  - Path values are always in a canonical form, so that you are relieved from trailing slashes, `.` and `..`.
+  - Path literals are automatically resolved [relative to the file](@docroot@/glossary.md#gloss-base-directory).
+  - Path values are automatically copied into the Nix store when used in a string interpolation or concatenation.
+  - Tooling can recognize path literals and provide additional features, such as autocompletion, refactoring automation and jump-to-file.
+
+  A path literal must contain at least one slash to be recognised as such.
+  For instance, `builder.sh` is not a path:
+  it's parsed as an expression that selects the attribute `sh` from the variable `builder`.
+
+  Path literals may also refer to absolute paths by starting with a slash.
+  This is generally not recommended, because it makes the expression less portable.
+  In the case where a path literal is translated into an absolute path string for a configuration file, it is recommended to just use strings.
+  This avoids some confusion about whether files at that location will be used during evaluation,
+  and it avoids unintentional situations where some function might try to copy everything at the location into the store.
```

Comment:
Usage notes should literally be notes. Otherwise it's hard to discern what is reference and what's chatter. Ideally we'd not have chatter at all, because the interfaces are accompanied by best-practice examples, which should prime users to do the right thing.

```suggestion

  > **Note**
  >
  > Absolute paths makes expressions less portable.
  > In the case where a path literal is translated into an absolute path string for a configuration file, it is recommended to just use strings.
  > This avoids some confusion about whether files at that location will be used during evaluation.
  > It also avoids unintentional situations where some function might try to copy everything at the location into the store.
```

I'm not sure what the "just use strings" thing is supposed to mean. You mean instead of `foo = /file.txt` do `foo = ''contents of file.txt''`?

Reply:
```suggestion

  > **Note**
  >
  > Absolute paths makes expressions less portable.
  > In the case where a function translates a path literal into an absolute path string for a configuration file, it is recommended to write a string literal instead.
  > This avoids some confusion about whether files at that location will be used during evaluation.
  > It also avoids unintentional situations where some function might try to copy everything at the location into the store.
```